### DISCUSSION TITLE
Multiply along a specific dimension

### DISCUSSION THREAD

**Discussion by adelezaini** [pydata/xarray#7066] [Q&A]
### Is your feature request related to a problem?

I was trying to make a simple operation for two DataArrays: applying the `*` operation along a single common dimension, but I couldn't find any simple and straight-forward way to apply it. This is a problem that can be faced very often when having weights to apply to a Dataset/DataArray, as in [this example](https://www.reddit.com/r/learnpython/comments/g45f2u/multiplying_xarray_dataarrays/).

### Describe the solution you'd like

I believe a very useful implementation would be adding `xr.mult(*arrays, dim, *args)` - or `xr.prod` or similar -, exactly as it is for the `xr.dot(*arrays, dim, *args)`, but instead of applying the `@` product it would apply the `*` product.

**Comment by benbovy** [comment]
Hi @adelezaini,

I'm a bit confused since you're trying to make a simple operation between two DataArrays and the example that you mention is about element-wise product of a DataArray with an unlabelled numpy array.

If you want to perform an element-wise operation like `*` between two DataArrays, xarray supports [automatic broadcasting by dimension name](https://docs.xarray.dev/en/stable/user-guide/computation.html#broadcasting-by-dimension-name), so all you need is to ensure that the two DataArrays share common dimension names and have the same number of elements along those dimensions (the latter is not strictly required if the DataArrays both have coordinates along those dimensions, in which case xarray also [automatically aligns](https://docs.xarray.dev/en/stable/user-guide/computation.html#automatic-alignment) the two DataArrays).

> I believe a very useful implementation would be adding xr.mult(*arrays, dim, *args) - or xr.prod or similar -,

There's already [DataArray.prod](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.prod.html) but it serves a different purpose, i.e., compute the product of array elements over a given dimension and return a reduced DataArray (like `numpy.prod`).

**Reply by adelezaini** [reply]
Thank you for aThank you for answering. 
The example I took was to illustrate the problem.
I also found another way to do it, but my point was suggesting to implement a very easy and intuitive method, analogous `xr.dot`. Even the broadcasting is not that straightforward to have a product along a specific axis, as it could be a function structured `xr.mult(*arrays, dim, *args)` specific to this task.
I suggest so because being a user of Xarray I expected it already implemented, since there are already analogous ones.

**Comment by benbovy** [comment]
@adelezaini I'm moving this issue to discussion Q&A as to my understanding Xarray already supports the operation that you describe here. Please let me know if I misunderstood your problem.
